components/formats-gpl/src/loci/formats/in/UnisokuReader.java

```diff
@@ -206,7 +211,7 @@ else if (key.startsWith(":y_data ->")) {
             pixelSizeY = Double.parseDouble(v[2]) - Double.parseDouble(v[1]);
             pixelSizeY /= getSizeY();
             if (unit.equals("nm")) {
-              pixelSizeY /= 1000;
+              pixelSizeYUnit = UNITS.NM;
```

Comment:
Removing the ```unit.equals("nm")``` checks and just using ```FormatTools.getPhysicalSizeX(pixelSizeX, unit)``` seems to make sense, since that will default to UNITS.MICROM anyway.